baremetal/src/Perf.cc

```diff
@@ -0,0 +1,213 @@
+//          Copyright Boston University SESA Group 2013 - 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <ebbrt/CpuAsm.h>
+#include <ebbrt/Cpuid.h>
+#include <ebbrt/Debug.h>
+#include <ebbrt/Perf.h>
+
+namespace {
+thread_local uint8_t pmcs{0};  // stores allocated gp counters
+};
+
+thread_local uint64_t ebbrt::perf::perf_global_ctrl{0};  // stored activated counters
```

Comment:
no need to put either of these two initializers. Zero is the default value for these already.